hashprobe/aeshash/aeshash_amd64.go

```diff
@@ -3,10 +3,34 @@
 package aeshash
 
 import (
-	"github.com/parquet-go/parquet-go/sparse"
+	"math/rand"
+	"unsafe"
+
 	"golang.org/x/sys/cpu"
+
+	"github.com/parquet-go/parquet-go/sparse"
 )
 
+// hashRandomBytes is 48 since this is what the assembly code depends on.
+const hashRandomBytes = 48
+
+var aeskeysched [hashRandomBytes]byte
+
+func init() {
+	key := (*[hashRandomBytes / 8]uint64)(unsafe.Pointer(&aeskeysched))
+	for i := range key {
+		key[i] = rand.Uint64()
+	}
+}
+
+// testingInitAesKeySched initializes the aes seed to a deterministic value for
+// test purposes.
+func testingInitAesKeySched() {
+	for i := range aeskeysched {
+		aeskeysched[i] = byte(i)
+	}
+}
```

Comment:
One way Go typically avoids having testing code in the package would be to make a `export_amd64_test.go` file that looks like this:

```go
//go:build amd64
package aeshash

func TestingInitAeskeysched() {
	for i := range aeskeysched {
		aeskeysched[i] = byte(i)
	}
}
```

then you can call `aeshash.TestingInitAeskeysched` from tests because the `*_test` files are included when building tests, and the function is part of the `aeshash` package.

Reply:
Yes, sorry, this is what I meant with "create _test.go files with compilation directives". Will do this then.